pkg/admission/interfaces.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package admission
+
+import (
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+)
+
+// Attributes is an interface used by AdmissionController to get information about a request
+// that is used to make an admission decision.
+type Attributes interface {
+	GetClient() client.Interface
```

Comment:
There are a number of options:

1. If we make Namespace as Kind, I want to use the client to fetch the Namespace object to ensure it exists before admitting the creation of an object in that namespace.
2. If we adopt the ```ResourceController``` proposal, I want to fetch the ResourceController in the namespace
3. In OpenShift, we may have downstream resources I want to read prior to making an admission decision.

At this point, I have a POC that uses it to implement the ResourceController proposal.